### PR TITLE
update envs to include pre-reqs

### PR DESCRIPTION
**Pull request by JaredGordon**
our env tests rely on existing resources (vpc, ig, policies, roles, keys etc) that are not included in the env templates we use for ci.

add these pre-reqs explicitly to the env templates, and use this as a way to help discover  pre-req layers (such as the "paperwork" layer).

### CODE REVIEW

File: ci/pipeline.yml [hunk 1]
```diff
@@ -3,7 +3,7 @@ resources:
 - name: terraforming
   type: git
   source:
-    branch: master
+    branch: envs
```
Comment (jvshahid) [inline ci/pipeline.yml]: @JaredGordon I think you forgot to undo this change.

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf [hunk 2]
```diff
@@ -1,71 +1,85 @@
...
 provider "aws" {
-  region     = "us-east-1"
+  region = "us-east-1"
 }
 
 module "providers" {
   source = "../../../../modules/dark_providers"
 }
 
+// pre-reqs
```
Comment (jvshahid) [inline env/dev/ops-sandbox/bastion-air-gapped/main.tf]: Is there a reason for not using the paperwork module here?

File: env/dev/ops-sandbox/bastion-vgw/main.tf [hunk 3]
```diff
@@ -9,15 +9,19 @@ terraform {
...
   region = "us-east-1"
   availability_zone = "us-east-1a"
+
   //hack to fix the path for windows, theoretically this will be fixed in v 0.12 to use same convention on all OS
   module_path = "${replace(path.module, "\\", "/")}"
+
   local_user_data_path = "${local.module_path}/other.yml"
-  inbound_ssh_cidrs = ["0.0.0.0/0"]
-  customer_ingress = { "443" = ["0.0.0.100/30", "0.0.1.0/28"], "8080" = ["0.0.0.128/26", "0.0.10.0/24"] }
+  inbound_ssh_cidrs = ["0.0.0.0/0"] customer_ingress = { "443" = ["0.0.0.100/30", "0.0.1.0/28"], "8080" = ["0.0.0.128/26", "0.0.10.0/24"] }
```
Comment (jvshahid) [inline env/dev/ops-sandbox/bastion-vgw/main.tf]: I think you accidentally joined two lines.  I am wondering if this is still valid hcl?

File: env/dev/ops-sandbox/bastion-vgw/main.tf [hunk 4]
```diff
@@ -28,10 +32,24 @@ module "providers" {
   source = "../../../../modules/dark_providers"
 }
 
+// pre-reqs
```
Comment (jvshahid) [inline env/dev/ops-sandbox/bastion-vgw/main.tf]: Same as before. Why aren't we using the paperwork module?

File: env/dev/ops-sandbox/bastion-vgw/main.tf [hunk 5]
```diff
@@ -9,15 +9,19 @@ terraform {
 }
 
 locals {
-  vpc_id = "vpc-0e83a6c1871b0afe2"
-  gw_id = "vgw-0e34270da663b10d4"
+  cidr_block = "10.0.0.0/24"
```
Comment (jvshahid) [inline env/dev/ops-sandbox/bastion-vgw/main.tf]: Can we call this local variable `vpc_cidr` to be consistent with the previous templates?

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf [hunk 6]
```diff
@@ -1,71 +1,85 @@
...
     kms_key_id = "7a0c75b1-b2e1-490d-8519-0aa44f1ba647"
     dynamodb_table = "bastion-air-gapped-state"
   }
 }
 
 locals {
-  vpc_id="vpc-00525b6ff996566ac"
-  region="us-east-1"
-  availability_zone="us-east-1a"
+  cidr_block = "10.0.0.0/24"
```
Comment (jvshahid) [inline env/dev/ops-sandbox/bastion-air-gapped/main.tf]: Can we call this local variable vpc_cidr to be consistent with the previous templates?

File: env/dev/vanilla/main.tf [hunk 7]
```diff
@@ -1,77 +0,0 @@
-terraform {
```
Comment (jvshahid) [inline env/dev/vanilla/main.tf]: Why are we deleting this?

File: modules/kms/lookup/main.tf [hunk 8]
```diff
@@ -1,13 +0,0 @@
-data "aws_kms_alias" "blobstore_kms_key_alias" {
```
Comment (jvshahid) [inline modules/kms/lookup/main.tf]: Why is file deleted?

File: modules/ops_manager/iam.tf [hunk 9]
```diff
@@ -1,7 +0,0 @@
-data "aws_iam_role" "ops_manager" {
```
Comment (jvshahid) [inline modules/ops_manager/iam.tf]: Why is this file deleted?  I think this change is unrelated to the story linked to in the commit message.  That aside,  I think there is some value in having the data lookup.  It serves as a verification for user input.  Not arguing whether it is valuable or not, but I think it is worth a discussion before removing it.

File: ci/pipeline.yml (hunk 1 above)
Comment (JaredGordon) [reply ci/pipeline.yml]: yeah: I'll set it back when we're done messing with our pipeline and it's back on master. It's a pain to keep checking it in, changing it back, checking it in again, etc.

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf (hunk 2 above)
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-air-gapped/main.tf]: we might want to split the paperwork layer into layers, since for bastion we don't need some of the stuff we need for PAS (iam stuff, keys...)

we are going to need to do something for the CP and shred services vpns, so we'll revisit then, and see if splitting it up makes sense.

File: env/dev/ops-sandbox/bastion-vgw/main.tf (hunk 3 above)
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-vgw/main.tf]: oops, I'll fix (tf didn't seem to mind for some reason...)

File: env/dev/ops-sandbox/bastion-vgw/main.tf (hunk 4 above)
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-vgw/main.tf]: we might in future (see above).

File: env/dev/ops-sandbox/bastion-vgw/main.tf (hunk 5 above)
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-vgw/main.tf]: yes, will fix.

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf (hunk 6 above)
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-air-gapped/main.tf]: yup!

File: env/dev/vanilla/main.tf (hunk 7 above)
Comment (JaredGordon) [reply env/dev/vanilla/main.tf]: this was used to test out the old terraforming-aws happy-path of public with route53 and so on. Since we're not supporting this anymore it's just extra work to keep it up to date.

File: modules/kms/lookup/main.tf (hunk 8 above)
Comment (JaredGordon) [reply modules/kms/lookup/main.tf]: using this data block caused dependency issues when we split out the paperwork layer. Deleting and replacing with a simple passed-in variable resolved this. The dependency is a hold-over from tf-aws.

File: modules/paperwork/main.tf [hunk 10]
```diff
@@ -0,0 +1,60 @@
+variable "vpc_cidr" {}
```
[the remark this answers is not in the document]
Comment (JaredGordon) [reply modules/paperwork/main.tf]: yeah, seems that way. For this PR I'd like to keep it as is, and then split when we get to the CP vpc, or when we do the peering (since we'll need it then). Just to not make it any more complex.

File: modules/paperwork/main.tf [hunk 11]
```diff
@@ -0,0 +1,60 @@
...
+
+variable "tags" {
+  type = "map"
+}
+
+locals {
```
[the remark this answers is not in the document]
Comment (JaredGordon) [reply modules/paperwork/main.tf]: name_prefix seemed to only allow six chars for us, and was ugly! 

We seemed to end up with vpcs that were named "air-gp34238y4983y5p9q8y54p8319y58p931y5p9831y4891y" which hurt my sensibilities.

This gives us pleasing names like "air-gapped-tuesday-rainbow-3476" (or whatever we want to call it).

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf (hunk 6 above)
[the remark this answers is not in the document]
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-air-gapped/main.tf]: agree, let's tackle when we do peering.

File: env/dev/ops-sandbox/bastion-vgw/main.tf (hunk 5 above)
[the remark this answers is not in the document]
Comment (JaredGordon) [reply env/dev/ops-sandbox/bastion-vgw/main.tf]: agree, let's tackle when we do peering.

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf (hunk 2 above)
Comment (Logiraptor) [reply env/dev/ops-sandbox/bastion-air-gapped/main.tf]: @JaredGordon If we decide to split it into layers, then we can replace these with those layers. It seems simpler today to reuse the same full layer in these spots, even if it's a little more than we strictly need.

File: ci/pipeline.yml (hunk 1 above)
Comment (jvshahid) [reply ci/pipeline.yml]: You don't/shouldn't have to do that.  We usually use `fly execute` to run modified concourse jobs.  This way the build history doesn't become confusing as you switch branches.

File: env/dev/vanilla/main.tf (hunk 7 above)
Comment (jvshahid) [reply env/dev/vanilla/main.tf]: Great.  :+1:  for deleting stuff that we don't need anymore.

File: modules/kms/lookup/main.tf (hunk 8 above)
Comment (jvshahid) [reply modules/kms/lookup/main.tf]: I like leaving the data block for the following reason.  Unlike other infrastructure inputs from the paperwork layer (such as vpc, subnet ids), having a typo in the KMS could be expensive to detect and have a long feedback cycle.  It will only fail when the job tries to use the key.  It is also harder to debug an error in the bosh job as opposed to getting an error from terraform saying that the key doesn't exist.  Furthermore, if all we are doing is passing through the value, why do we even need the module?

p.s. It could be the case that OM has a KMS key verifier, which could replace the usage of data blocks to verify the input.

File: modules/ops_manager/iam.tf (hunk 9 above)
Comment (jvshahid) [reply modules/ops_manager/iam.tf]: Resolving this thread since we are talking about the same concern in [this thread](https://github.com/pivotal-federal-field-engineering/terraforming-pws-dark/pull/23#discussion_r284760172)

File: ci/pipeline.yml (hunk 1 above)
Comment (JaredGordon) [reply ci/pipeline.yml]: good point!

File: modules/kms/lookup/main.tf (hunk 8 above)
Comment (JaredGordon) [reply modules/kms/lookup/main.tf]: makes sense: we're going to run into these hidden coupled dependencies as we decouple stuff into layers. Fritz and I are going to have to spend some time teasing these out. terraforming pas is way coupled right now. I'll create a story to define an infra layer (as distinct from the pas layer).

File: modules/paperwork/main.tf (hunk 11 above)
[the remark this answers is not in the document]
Comment (JaredGordon) [reply modules/paperwork/main.tf]: as discussed, we will remove the "random" code from the paperwork module, it belongs in the ci envs.

File: env/dev/ops-sandbox/bastion-air-gapped/main.tf (hunk 2 above)
Comment (jvshahid) [reply env/dev/ops-sandbox/bastion-air-gapped/main.tf]: Resolved this since we started working on the staging environment which will make some of those environments obsolete.
